lib/puppet/parser/scope.rb

```diff
@@ -738,17 +815,27 @@ def to_s
 
   alias_method :inspect, :to_s
 
-  # remove ephemeral scope up to level
-  # TODO: Who uses :all ? Remove ??
```

Comment:
There is not a single use case of unset_ephemeral_var(:all) and if someone called that it is a sure way of creating havoc. Suggest not changing the implementation to correctly return the popped ephemeral scopes. I would much rather see a deprecation warning.